test/Microsoft.AspNet.Mvc.Core.Test/DefaultAssemblyProviderTests.cs

```diff
@@ -98,6 +100,51 @@ public void CandidateAssemblies_ReturnsEmptySequenceWhenReferenceAssembliesIsNul
             Assert.Empty(nullProviderCandidates.Select(a => a.Name));
         }
 
+        // This test verifies DefaultAssemblyProvider.ReferenceAssemblies returns the assembly list that we intended.
+        // If ReferenceAssemblies needs to be changed, the expected list in this test should be updated together.
+        [Fact]
+        public void ReferenceAssemblies_ReturnsExpectedReferenceAssemblies()
```

Comment:
And, for whatever reason, the assemblies in the unit test are listed in a different order from the product code (not that it matters, because the sets are equal, but it makes me even more suspicious about this).